src/main/java/teammates/common/datatransfer/FeedbackConstantSumQuestionDetails.java

```diff
@@ -450,14 +475,38 @@ public String getQuestionResultStatisticsCsv(
 
         DecimalFormat df = new DecimalFormat("#.##");
         
+        Map<String, List<Integer>> sortedOptionPoints = new TreeMap<String, List<Integer>>();
+        
+        Map<String, String> identifierMap = new HashMap<String, String>();
+        
         for (Entry<String, List<Integer>> entry : optionPoints.entrySet()) {
+            
+            if (distributeToRecipients) {
+                
+                String participantIdentifier = entry.getKey();
+                String name = bundle.getNameForEmail(participantIdentifier);
+                String nameEmail = name + participantIdentifier;
+                
+                identifierMap.put(nameEmail, participantIdentifier);
+                sortedOptionPoints.put(nameEmail, entry.getValue());
+                
+            } else {
+                
+                String option = options.get(Integer.parseInt(entry.getKey()));
+                
+                identifierMap.put(option, entry.getKey());
+                sortedOptionPoints.put(option, entry.getValue());
+            }
+        }
+        
+        for (Entry<String, List<Integer>> entry : sortedOptionPoints.entrySet()) {
             String option;
             if (distributeToRecipients) {
-                String teamName = bundle.getTeamNameForEmail(entry.getKey());
-                String recipientName = bundle.getNameForEmail(entry.getKey());
+                String teamName = bundle.getTeamNameForEmail(identifierMap.get(entry.getKey()));
+                String recipientName = bundle.getNameForEmail(identifierMap.get(entry.getKey()));
```

Comment:
@JosephineKwa Everything is done, just that I have to resolve the merge conflict and regenerate the test files because I let it overwrite my expected test files as manually resolving the expected htmls are too time consuming. Will let you know when it is done:)